### PR TITLE
Support XmlDownload.aspx

### PR DESCRIPTION
**Pull request by abrampl**

### CODE REVIEW

**Review by abrampl — COMMENTED** [review]
Fix new entry line

File: src/Picqer/Financials/Exact/Connection.php [hunk 1]
```diff
@@ -537,6 +559,29 @@ private function parseResponseXml(Response $response)
...
+     */
+    private function parseDownloadResponseXml(Response $response)
+    {
+        try {
+            if ($response->getStatusCode() === 204) {
+                return [];
+            }
+
+            Psr7\Message::rewindBody($response);
+            $simpleXml = new \SimpleXMLElement($response->getBody()->getContents());
```
Comment (faizanakram99) [inline src/Picqer/Financials/Exact/Connection.php:577]:
```suggestion
            $simpleXml = new \SimpleXMLElement($response->getBody()->__toString());
```

If `__toString()` is used, there is no need to manually rewind stream. `StreamInterface::getContents()` behaviour is inconsistent across different psr 7 libraries.

https://www.php-fig.org/psr/psr-7/#13-streams

**Review by faizanakram99 — CHANGES_REQUESTED** [review]

**Review by faizanakram99 — APPROVED** [review]

**Comment by stephangroen** [comment]
Any explanation on what this is?

**Comment by faizanakram99** [comment]
> Any explanation on what this is?

@stephangroen 
From https://support.exactonline.com/community/s/knowledge-base#All-All-DNO-Content-xmlintro
> Exact Online XML API - Introduction
> In Exact Online you can manually import and export data using the XML API. You can also exchange data from external applications automatically. This allows Exact Online to integrate seamlessly with solutions developed by external software suppliers.
> 
> The Exact Online XML API can easily be integrated in external applications by requesting the XMLDownload.aspx or the XMLUpload.aspx and specifying the topic which needs to be downloaded or uploaded with some optional parameters.
> 
> For all API requests, you are required to authenticate the request.

Some code examples here
https://support.exactonline.com/community/s/knowledge-base#All-All-DNO-Content-xmlsamplecode

**Comment by mvdgeijn** [comment]
Accepting this pull request would be much appreciated. I was already working on the same.

**Review by stephangroen — CHANGES_REQUESTED** [review]
Thanks. Please fix the one line failing in the style, and I'll merge the PR.

**Comment by stephangroen** [comment]
Fixed in #586

**Pull request closed without merging**